climpred/classes.py

```diff
@@ -165,12 +267,20 @@ def smooth(self, smooth_kws='goddard2013'):
             raise ValueError(
                 'Please provide kwargs as str or dict and not', type(smooth_kws)
             )
-        self.initialized = smooth_fct(self.initialized, smooth_kws)
-        # check for other objects in  PredictionEnsemble
-        # better: for obj in self: do smooth
-        if isinstance(self.uninitialized, xr.Dataset):
-            self.uninitialized = smooth_fct(self.uninitialized, smooth_kws)
-        # self.reference not implemented
+        # Apply throughout the dataset
+        # TODO: Parallelize
+        datasets = self._datasets.copy()
+        datasets['initialized'] = smooth_fct(self._datasets['initialized'], smooth_kws)
+        # Apply if uninitialized, control, reference exist.
+        if self._datasets['uninitialized']:
+            datasets['uninitialized'] = smooth_fct(
+                self._datasets['uninitialized'], smooth_kws
+            )
+        if isinstance(self, PerfectModelEnsemble):
+            if self._datasets['control']:
+                datasets['control'] = smooth_fct(self._datasets['control'], smooth_kws)
+        # if type(self).__name__ == 'HindcastEnsemble':
```

Comment:
Is this still necessary?

Reply:
I have it commented out because it was something I wanted to check on. The current `.smooth()` doesn't run cleanly right now with the objects. It'll need a bit of refactoring/modifying but I'm going to open an issue to address that in another PR before the next release.